src/mesh.cpp

```diff
@@ -79,25 +79,31 @@ void readMesh(Mesh &mesh, const std::string &filename, const int dim)
     mesh.positions.push_back(vertexLoc);
   }
 
-  for (int i = 0; i < grid->GetNumberOfCells(); i++) {
-    int cellType = grid->GetCell(i)->GetCellType();
-
-    // Here we use static cast since VTK library returns a long long unsigned int however preCICE uses int for PointId's
-    if (cellType == VTK_TRIANGLE) {
-      vtkCell                 *cell = grid->GetCell(i);
-      std::array<Mesh::VID, 3> elem{vtkToPos(cell->GetPointId(0)), vtkToPos(cell->GetPointId(1)), vtkToPos(cell->GetPointId(2))};
-      mesh.triangles.push_back(elem);
-    } else if (cellType == VTK_LINE) {
-      vtkCell                 *cell = grid->GetCell(i);
-      std::array<Mesh::VID, 2> elem{vtkToPos(cell->GetPointId(0)), vtkToPos(cell->GetPointId(1))};
-      mesh.edges.push_back(elem);
-    } else if (cellType == VTK_QUAD) {
-      vtkCell                 *cell = grid->GetCell(i);
-      std::array<Mesh::VID, 4> elem{vtkToPos(cell->GetPointId(0)), vtkToPos(cell->GetPointId(1)), vtkToPos(cell->GetPointId(2)), vtkToPos(cell->GetPointId(3))};
-      mesh.quadrilaterals.push_back(elem);
-    } else {
-      std::cerr << "Invalid cell type in VTK file. Valid cell types are, VTK_LINE, VTK_TRIANGLE, and VTK_QUAD.";
-      MPI_Abort(MPI_COMM_WORLD, EXIT_FAILURE);
+  if (requireConnectivity) {
+    for (int i = 0; i < grid->GetNumberOfCells(); i++) {
+      int cellType = grid->GetCell(i)->GetCellType();
+
+      // Here we use static cast since VTK library returns a long long unsigned int however preCICE uses int for PointId's
+      if (cellType == VTK_TRIANGLE) {
+        if (dim == 3) {
+          vtkCell                 *cell = grid->GetCell(i);
+          std::array<Mesh::VID, 3> elem{vtkToPos(cell->GetPointId(0)), vtkToPos(cell->GetPointId(1)), vtkToPos(cell->GetPointId(2))};
+          mesh.triangles.push_back(elem);
+        }
+      } else if (cellType == VTK_LINE) {
+        vtkCell                 *cell = grid->GetCell(i);
+        std::array<Mesh::VID, 2> elem{vtkToPos(cell->GetPointId(0)), vtkToPos(cell->GetPointId(1))};
+        mesh.edges.push_back(elem);
+      } else if (cellType == VTK_QUAD) {
+        if (dim == 3) {
+          vtkCell                 *cell = grid->GetCell(i);
+          std::array<Mesh::VID, 4> elem{vtkToPos(cell->GetPointId(0)), vtkToPos(cell->GetPointId(1)), vtkToPos(cell->GetPointId(2)), vtkToPos(cell->GetPointId(3))};
+          mesh.quadrilaterals.push_back(elem);
+        }
+      } else {
+        std::cerr << "Invalid cell type in VTK file. Valid cell types are, VTK_LINE, VTK_TRIANGLE, and VTK_QUAD.";
+        MPI_Abort(MPI_COMM_WORLD, EXIT_FAILURE);
+      }
```

Comment:
Let's also add a `VLOG` after this if block: `Skipping the setup of connectivity information for mesh <Name> as no connectivity is required`

Reply:
Done